speechbrain/utils/distributed.py

```diff
@@ -103,8 +94,13 @@ def main_process_only(function):
     @wraps(function)
     def main_proc_wrapped_func(*args, **kwargs):
         """This decorated function runs only if this is the main process."""
+        os.environ[MAIN_PROC_ENV] = "1"
```

Comment:
Additionally I wonder if the environment variables (like `MAIN_PROC_ENV` here) are the right way to do this sort of process-wide communication. I think something like a variable in a module should be enough here. So instead of this, I think we could just have:

```python
MAIN_PROC_FLAG=0

def main_proc_wrapped_func(*args, **kwargs):
    global MAIN_PROC_FLAG
    MAIN_PROC_FLAG = 1
    ...
    MAIN_PROC_FLAG = 0
    
    
def ddp_barrier():
    global MAIN_PROC_FLAG  # Actually I think this is unnecessary even.
    if MAIN_PROC_FLAG == 0:
        ...
```

Reply:
Ah yes, a module-level flag is better here.